docs/articles/documentation/extending-testcafe/custom-reporter/creating-reporter-plugin/index.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: docs
+title: Creating Custom Reporter
+permalink: /documentation/extending-testcafe/custom-reporter-plugin/creating-reporter-plugin/
+---
+# Creating Reporter Plugin
+
+## Generating Reporter Project
+
+First, install [Yeoman](http://yeoman.io) and `generator-testcafe-reporter` using [npm](https://www.npmjs.com/) (we assume you have pre-installed [node.js](https://nodejs.org/)).
+
+```bash
+npm install -g yo
+npm install -g generator-testcafe-reporter
+```
+
+Create a new directory where the generator should place your scaffolded project files and go into it.
+
+```bash
+mkdir MyReporter
+cd MyReporter
+```
+
+Then run the reporter generator to create a new project:
+
+```bash
+yo testcafe-reporter
+```
+
+The generator will provide optional settings to customize your reporter. You need to specify:
+
+* a reporter name. The generator will suggest the reporter name that matches the reporter directory name. You can keep this name or specify another one.
+* your GitHub username.
+* the URL of your website.
+* whether you want to implement a custom error decorator for the reporter.
```

Comment:
This is the only place where we mention a *custom error decorator* (I searched for the word 'decorator') =))
We need to explain it somehow.